src/main/scala/kamon/datadog/DatadogAgentReporter.scala

```diff
@@ -87,12 +95,12 @@ class DatadogAgentReporter private[datadog] (c: DatadogAgentReporter.Configurati
   private def encodeDatadogCounter(count: Long, unit: MeasurementUnit): String =
     valueFormat.format(scale(count, unit)) + "|c"
 
-  private def encodeDatadogGauge(value: Long, unit: MeasurementUnit): String =
+  private def encodeDatadogGauge(value: Double, unit: MeasurementUnit): String =
     valueFormat.format(scale(value, unit)) + "|g"
 
-  private def scale(value: Long, unit: MeasurementUnit): Double = unit.dimension match {
-    case Time if unit.magnitude != time.seconds.magnitude             => MeasurementUnit.scale(value, unit, time.seconds)
```

Comment:
Not sure why this was hard coded to seconds. But it was causing times to report incorrectly

Reply:
I'm almost sure that at some point in time it was assumed that everything that gets to the agent had to be in seconds. If that is no longer the case then all good with that!

Reply:
Hmmm. Well my current setup actually posts to telegraf, and telegraf/grafana _definitely_ doesn't like seconds. Let me check actual datadog, but my hunch is this is incorrect

Reply:
You know, thinking about it a bit more, does it really matter? as long as the unit used is the one in the configuration we are all good! the only important thing would be have the default value in the reference config to be whatever Datadog expects by default.

Reply:
Yeah. It now correctly respects the configured value, whereas before that value was never used.